adaptive/runner.py

```diff
@@ -232,6 +248,12 @@ def _get_futures(self):
             fut = self._submit(x)
             fut.start_time = start_time
             self.pending_points[fut] = x
+            try:
+                i = _key_by_value(self._index_to_point, x)  # O(N)
+            except StopIteration:  # `x` is not a value in `self._index_to_point`
+                self._i += 1
+                i = self._i
+            self._index_to_point[i] = x
```

Comment:
This logic should be placed in `self._ask` instead, and can be simplified by having an ID-generating state machine as I show above

Reply:
Fixed in 62e9a416a964f43ab79b78c1463f67c04e28633e.